chess2.cpp

```diff
@@ -0,0 +1,882 @@
+
+        if (row > 0 && col > 1 && board[row - 1][col - 2] <= 0) // one up two left
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col - 2));
+    }
+
+    else if (!turn) {
+        if (row > 0 && col < 6 && board[row - 1][col + 2] >= 0)pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col + 2));
+        if (row > 1 && col < 7 && board[row - 2][col + 1] >= 0)pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 2, col + 1));
+        if (row < 7 && col < 6 && board[row + 1][col + 2] >= 0)pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col + 2));
+        if (row < 6 && col < 7 && board[row + 2][col + 1] >= 0)pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 2, col + 1));
+        if (row < 6 && col > 0 && board[row + 2][col - 1] >= 0)pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 2, col - 1));
+        if (row < 7 && col > 1 && board[row + 1][col - 2] >= 0)pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col - 2));
+        if (row > 1 && col > 0 && board[row - 2][col - 1] >= 0)pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 2, col - 1));
+        if (row > 0 && col > 1 && board[row - 1][col - 2] >= 0)pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col - 2));
+    }
+
+    //returnpseudomoves;
+}
+
+void Chess2::getpawnmoves(bool turn, int row, int col) {
+    if (turn) {
+        if (row == 0){
+            return ;
+        }
+        if (row == 6 && board[row - 1][col] == 0 && board[row - 2][col] == 0)
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 2, col));
+        if (board[row - 1][col] == 0)
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col));
+        if (col != 0 && board[row - 1][col - 1] < 0)
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col - 1));
+        if (col != 7 && board[row - 1][col + 1] < 0)
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col + 1));
+    }
+
+    else if (!turn) {
+        if (row == 7){
+            return ;
+        }
+        if (row == 1 && board[row + 1][col] == 0 && board[row + 2][col] == 0){
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 2, col));
+        }
+        if (board[row + 1][col] == 0)
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col));
+        if (col != 0 && board[row + 1][col - 1] > 0)
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col - 1));
+        if (col != 7 && board[row + 1][col + 1] > 0)
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col + 1));
+    }
+
+    //returnpseudomoves;
+}
+
+void Chess2::getkingmoves(bool turn, int row, int col) {
+
+    if (!turn) {
+        if (row != 7 && board[row + 1][col] >= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col));
```

Comment:
Instead of writing so many conditions, we could have use array which gives us next possible move. 
Create a different function to check for possible move.